.github/workflows/build_main.yml

```diff
@@ -24,6 +24,11 @@ on:
     branches:
     - '**'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  # only cancel in forked repositories
+  cancel-in-progress: ${{ github.repository_owner != 'apache' }}
```

Comment:
Okay, there are two issues :-).

1. With `concurrency`, it introduces a new state like pending:

![Uploading Screenshot 2024-05-22 at 8.17.03 PM.png…]()

and the notify job fails at https://github.com/apache/spark/blob/master/.github/workflows/notify_test_workflow.yml#L122

So increasing the timeout here a lot would help.

2. The builds in the master branch in this main repo became sequential. It doesn't cancel and wait for the build in the previous commit to finish.